src/v/resource_mgmt/scheduling_groups_probe.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/partition_leaders_table.h"
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+#include "resource_mgmt/cpu_scheduling.h"
+#include "ssx/metrics.h"
+
+#include <seastar/core/metrics.hh>
+
+class scheduling_groups_probe {
+public:
+    void wire_up(const scheduling_groups& scheduling_groups) {
+        if (config::shard_local_cfg().disable_public_metrics()) {
+            return;
+        }
+
+        auto groups = scheduling_groups.all_scheduling_groups();
+        for (auto&& group_ref : groups) {
+            _public_metrics.add_group(
+              prometheus_sanitize::metrics_name("scheduler"),
+              {seastar::metrics::make_gauge(
```

Comment:
I can never remember the naming convention, but I think this is a counter, and have a  `_total` suffix?

Reply:
Yep. I think you're right. It can only increase so it should be a counter.

Reply:
Addressed.